### DISCUSSION TITLE
Frame sender fields

### DISCUSSION THREAD

**Discussion by exekutive** [collin80/SavvyCAN#384] [Q&A] [answered]
What are the "Ext" & "Rem" columns in the 'Custom' or 'Frame Sender' utility?

**Comment by collin80** [comment, marked as answer]
Extended and Remote. Extended frames have 29 bit ID values. Remote frames have no data bytes but do specify a length. They are used to request that the receiver send back a message with that length. Remote frames are deprecated and you won't often find them used.

**Reply by exekutive** [reply]
Thanks, and it looks SavvyCAN will set 'ext' automatically if you enter an ID long enough.

**Reply by jogo-** [reply]
Yes, correct. The 'ext' is set when the ID is > 0x7FF, as you can see in code:
```
if (frame.frameId() > 0x7FF) frame.setExtendedFrameFormat(true);
...
thisFrame.setExtendedFrameFormat(thisFrame.frameId() > 0x7FF);
```
which can be problematic for extended frames (29bits) but with ID <= 0x7FF ...
